src/Compilers/Core/Portable/Syntax/SyntaxList.SeparatedWithManyChildren.cs

```diff
@@ -39,6 +37,21 @@ internal SeparatedWithManyChildren(InternalSyntax.SyntaxList green, SyntaxNode?
 
                 return _children[i >> 1].Value;
             }
+
+            internal override int GetChildPosition(int index)
+            {
+                // If the previous sibling (ignoring separator) is not cached, but the next sibling
+                // (ignoring separator) is cached, use the next sibling to determine position.
+                int valueIndex = (index & 1) != 0 ? index - 1 : index;
+                if (valueIndex > 1
+                    && _children[(valueIndex - 2) >> 1].Value is null
+                    && (valueIndex >= Green.SlotCount - 2 || _children[(valueIndex + 2) >> 1].Value is { }))
```

Comment:
>valueIndex >= Green.SlotCount - 2 [](http://example.com/codeflow?start=24&length=33)

Could you elaborate what is the meaning of this condition?

Reply:
The preceding conditions have determined the previous sibling is not cached. This condition checks if the index represents the last item or the next to last item. If so, we use `GetChildPositionFromEnd()` to check following siblings. If the index is the last item, that's the base case in `GetChildPositionFromEnd()` where there are no following siblings. If it's the next to last item, the only sibling is the last separator, so that's also a simple base case.

Reply:
I think it would be good capturing this in a comment and explaining why it doesn't matter whether the next item is cached in this case.